jib-core/src/main/java/com/google/cloud/tools/jib/http/FailoverHttpClient.java

```diff
@@ -97,7 +101,17 @@ private static HttpTransport getSecureHttpTransport() {
     //
     // A new ApacheHttpTransport needs to be created for each connection because otherwise HTTP
     // connection persistence causes the connection to throw NoHttpResponseException.
-    return new ApacheHttpTransport();
+    HttpClientBuilder httpClientBuilder =
+        HttpClientBuilder.create()
+            .useSystemProperties()
+            .setSSLSocketFactory(SSLConnectionSocketFactory.getSystemSocketFactory())
+            .setMaxConnTotal(200)
+            .setMaxConnPerRoute(20)
+            .setConnectionTimeToLive(-1, TimeUnit.MILLISECONDS)
+            .setRoutePlanner(new SystemDefaultRoutePlanner(ProxySelector.getDefault()))
+            .disableRedirectHandling()
+            .disableAutomaticRetries();
+    return new ApacheHttpTransport(httpClientBuilder.build());
```

Comment:
Thanks for your contribution!

As in `getInsecureHttpTransport()` below, can't this just be
```java
          ApacheHttpTransport.newDefaultHttpClientBuilder()
              .setSSLSocketFactory(SSLConnectionSocketFactory.getSystemSocketFactory());
```
? This is exactly [the reason for asking Google HTTP Client to add `newDefaultHttpClientBuilder()`](https://github.com/googleapis/google-http-java-client/issues/578) in the past.

Reply:
Yes, that would work and is cleaner. I applied your suggestion.